### ISSUE TITLE
Cluster health-check may cause applications inconsistencies

### ISSUE THREAD

**Issue by rymsha** [enonic/xp#8086]
When cluster health gets from healthy to unhealthy and back to healthy lots of OSGi services are re-created. But a some logic around Applications is done on Bundle level (with BundleTrackers/Customizers/Listeners), and they are not necessarily invalidated.

**Comment by rymsha** [comment]
#8114 improves live a bit, but Unhealthy cluster still turns off way too much:

(ES)`Client` stops -> `ApplicationServiceImpl` stops 
-> `AutoDeployer`, `DeployDirectoryWatcher`, `StoredApplicationsDeployer` stop -> `ApplicationDeployerManager` stops
-> ScriptRuntimeFactoryImpl stops (and don't call invalidators anymore)

when `Client` is back again `ApplicationDeployerManager` activates and "reinstalls" all applications - it is not harmful by itself, but `disposers` (and maybe some other livalidators) won't be called for old application instantiations.

**Comment by rymsha** [comment]
I'm starting to think that Cluster HealthCheck service was a bad idea in general.

- it leads OSGi services to be created in parallel threads on XP startup, which is deadlock-prone #6517 #8056

- it stops applications during downtime and makes writing monitoring applications impractical #7953

- it is hard to tell when cluster is exactly "unhealthy" #6700 

- elasticseach cluster RED does not mean it is fully nonfunctional, but disconnecting from it during maintenance could easily make in so #7328

- stored application get "installed" second time when health-check is satisfied (#8086 this issue)

**Comment by sigdestad** [comment]
With No Ignite, we can remove it alltogether imho

**Comment by rymsha** [comment]
Removed separate thread for health-check.

Good news is that XP starts and works.

Bad news: SnapshotService is waiting for health-check to report RED (regardless of why RED), to start (😮) snapshot restore.  
Also some statuses don't work correctly. 

Will have to go deeper in than rabbit hole, @gbbirkisson

**Comment by rymsha** [comment]
enonic/cli-enonic#223

**Comment by rymsha** [comment]
#6989

**Comment by rymsha** [comment]
#5831

**Comment by rymsha** [comment]
Removal of health-check revealed a few more issues in Initializer implementations
1. All Initialzers (i.e. SystemRepoInitializer) may run too early - before elasticsearch can accept any operations on indices. This I can fix by waiting for yellow status individually in every initializer
2. Most of initializers report (isInitializerd == true) little bit too early. Causing races like #6681

**Comment by rymsha** [comment]
#8172 is yet another issue that appears more often in health-check is removed.
